test/unit/service_discovery/cluster_client_test.rb

```diff
@@ -197,5 +180,38 @@ class ClusterClientTest < ActiveSupport::TestCase
         cluster.find_project_by(name: 'my-namespace')
       end
     end
+
+    class BuildClientTest < ActiveSupport::TestCase
+      setup do
+        @config = {
+          server_scheme: 'https',
+          server_host: 'my-cluster.com',
+          server_port: 8443,
+          bearer_token: 'secret-token'
+        }
+      end
+
+      test '.build_client' do
+        assert_equal client_endpoint, 'https://my-cluster.com:8443/api'
+        assert_equal client_endpoint(api_path: 'k8s'), 'https://my-cluster.com:8443/k8s'
+      end
+
+      test '.build_client_for_projects' do
+        assert_equal client_endpoint_for_projects, 'https://my-cluster.com:8443/apis/project.openshift.io'
+        assert_equal client_endpoint_for_projects(api_path_for_projects: 'projs'), 'https://my-cluster.com:8443/projs'
+        assert_equal client_endpoint_for_projects(api_path: 'k8s'), 'https://my-cluster.com:8443/k8s'
+        assert_equal client_endpoint_for_projects(api_path: 'k8s', api_path_for_projects: 'projs'), 'https://my-cluster.com:8443/projs'
```

Comment:
Is this correct? Wouldn't it be `https://my-cluster.com:8443/k8s/projs`

Reply:
It is correct. The idea is to show that `api_path_for_projects` takes precedence and overrides `api_path` if both are provided.

Reply:
Ok. Removed `api_path` from `.build_client_for_projects` for simplicity.